### ISSUE TITLE
Keep color and formatting information in Excel

### ISSUE THREAD

**Issue by chankeypathak** [DataTables/TableTools#44]
If you download the Excel file of table it shows all the content correctly. Kindly add the feature of keeping the formatting and colour information of cells so that the table looks exactly the same as it looks on webpage. 

I achieved this without using TableTools here: http://jsfiddle.net/chankeypathak/7cwZ9/1/show/

But I really want to use TableTools. Can you please add this feature? Thanks.

**Comment by jklaus** [comment]
I have actually written a library that implements xlsx.js (for FF & Chrome) and ActiveX (calling Excel.Application, for IE) to support well formatted XLSX export from json.  I believe both of these approaches support support cell level styling.. I will actually be digging into that within the next couple days.  I was going to make it accessible via an extension/prototype.. but if you want I could add it into the primary table tools.

**Comment by p0tvin29** [comment]
@jklaus are you able to provide your library?

**Comment by jklaus** [comment]
@p0tvin29 yes, take a look at <a href="https://github.com/jklaus/jk">jk</a>

I haven't gotten back around to adding cell level styling though.  Given we can probably just drop IE8 (ActiveX) support it should be fairly painless if memory serves.  I also have an extension to DataTables that I wrote which provides an "Excel" button to TableTools.  It can be overridden to create complex exports or simply implemented and the exporter reflects on the current column options to hook everything up... if you don't get a message back from me in regard to this piece tomorrow feel free to bump this thread.

**Comment by jklaus** [comment]
I went ahead and added a crude version of the file.. see TableTools.Extensions.js

**Comment by DataTables** [comment]
I should point out that TableTools has be retired in favour of [Buttons](/DataTables/Buttons). It will export true xlsx files so you might find it easier to add formatting there.

**Comment by jklaus** [comment]
Assuming folks aren't stuck in legacy correct?

**Comment by DataTables** [comment]
Yes. I'm no longer actively developing TableTools - but its open source so it can continue if needed!

**Comment by jklaus** [comment]
At the time I was stuck in legacy - had to support IE8, IE11, Chrome, FF, Opera, etc.  Now we're finally off IE8.

**Comment by xezpeleta** [comment]
I need something similar. I would like to export my table to Excel, **keeping the links**.

For instance, I have a table with the following column:

```html
<td><a href="about:blank">This is a link</a></td>
```

When I export my table to Excel, I only can see "_This is a link_" but it's not linked.

Do you know if is it possible to do that?

Thanks in advance

**Comment by BeingWasiZaidi** [comment]
export to xlsx with column background color ?

**Comment by Shaheerarain786** [comment]
@BeingWasiZaidi 
`"buttons": ['pageLength',
                        {
                            extend: 'excel',
                            text: 'Export excel',
                            footer:true,
                            customize: function( xlsx ) {
                                var sheet = xlsx.xl.worksheets['sheet1.xml'];
                                $('row c[r^="G"]', sheet).each( function () {
                                    if ( $(this).text()>= 1) {
                                        $(this).attr( 's', '56' );                 
                                    }
                                });
                            },
                            exportOptions: {                               
                                 columns: ':visible',                              
                            }
                        }
                    ],`
This is how you can change the background color of your cell while exporting the excel file. 
Regards.
